src/components/CollectButton.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div class="flex flex-col gap-[8px] justify-center items-center">
+    <ButtonV2
+      preset="secondary"
+      :is-disabled="!isCollectable"
+      @click="handleClickCollectButton"
+    >
+      <template v-if="isCollectable" #prepend>
+        <IconPrice />
+      </template>
+      {{ buttonText }}
+    </ButtonV2>
+    <div 
+      v-if="collectExpiryTimeForDisplay 
+        && (isCollectExpiryTimeNear || displayNonHighlightedExpiryTime)" 
+      class="flex gap-[4px] justify-center items-center" 
+      :class="expiryTimeClass"
+    >
+      <IconClock />
+      <div class="ml-[4px]">{{ collectExpiryTimeForDisplay }}</div>
+    </div>
+  </div>
+</template>
+
+<script>
+import dayjs from 'dayjs';
+
+import { normalizeLocaleForDayjs } from '~/locales';
+
+export default {
+  props: {
+    buttonText: {
+      type: String,
+      default: '',
+    },
+    isCollectable: {
+      type: Boolean,
+      default: false,
+    },
+    collectExpiryTime: {
+      type: Number,
+      default: 0,
+    },
+    displayNonHighlightedExpiryTime: {
```

Comment:
How about
```suggestion
    shouldAlwaysDisplayExpiryTime: {
```

Reply:
Ever thinking of calling the prop with the term `always`, but the passed expiry time will not display in our case.
Is it OK to name it `shouldAlwaysDisplayExpiryTime`?

Reply:
Then how about
```suggestion
    shouldShowExpiryTimeBeforeExpired: {
```